module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,324 @@
+                        [
+                            { "$match": { "userId": userId, isDeleted: false } },
+                            {
+                                "$project": {
+                                    '_id': "$pollId",
+                                    'name': '$pollName'
+                                }
+                            },
+                            { "$sort": { createdAt: 1 } }
+                        ])
+            ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+                          
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                          result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                          result = result.concat(response[1]);
+                        }
+                    });
+
+
+                if (!result.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: _.uniqWith(result, _.isEqual)
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+
+    /**
+    * Validate poll submission.
+    * @method
+    * @name validatePollSubmission
+    * @param {String} pollId - pollId
+    * @param {String} userId - userId
+    * @returns {String} - poll status and poll submission document.
+    */
+
+   static validatePollSubmission(pollId= "", userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if (pollId == "") {
+                throw new Error(messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK)
+            }
+
+            if (userId == "") {
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+            }
+
+            let pollDocument = await pollsHelper.pollDocuments
+            (
+                { _id: pollId,
+                  isDeleted: false },
+                [
+                    "status",
+                    "endDate"
+                ]
+            )
+            
+            if (!pollDocument.length) {
+                throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+            }
+
+            let result = {
+                pollStatus: pollDocument[0].status,
+                pollSubmissionDocument: []
+            }
+
+            if (new Date() > new Date(pollDocument[0].endDate)) {
+                result.pollStatus = messageConstants.apiResponses.INACTIVE_STATUS
```

Comment:
@deepakr123 - Date condition can be checked in query object itself.

Reply:
resolved